### PR TITLE
Add release docs

### PR DESCRIPTION
**Pull request by jamwaffles**
Astute reviewers will recognise most of the e-g release steps in here, but they're plenty good enough to copy over ;)

### CODE REVIEW

**Comment by jamwaffles** [comment]
When this PR is merged, it would be great if @rfuest could try releasing 0.3.1 to publish #34 and exercise the release docs.

**Comment by rfuest** [comment]
> When this PR is merged, it would be great if @rfuest could try releasing 0.3.1 to publish #34 and exercise the release docs.

Good idea. I don't have time in the moment, but I'll try to find some time in the next few days.

We could also merge this after I've released `0.3.1`, so that we can tweak the docs if I encounter any problems.

**Comment by jamwaffles** [comment]
> I don't have time in the moment, but I'll try to find some time in the next few days.

Awesome, let us know when you get round to it.

> We could also merge this after I've released 0.3.1, so that we can tweak the docs if I encounter any problems.

Also a good idea. If I could find it, I'd allow you to push to this branch but I can't find the setting...

**Comment by jamwaffles** [comment]
By the way, I've added the e-g core team as allowed to push to `master` for tinytga, tinybmp, simulator and e-g itself so when you release, hopefully the Git tags are pushed by `cargo-release`. I'd suggest attempting to release one of the `tiny*` crates first to figure out any teething problems.

File: doc/releasing.md
Comment (rfuest) [inline doc/releasing.md]:
```suggestion
  cargo release --push-remote <push-remote> <level>

  Where `<level>` is `major`, `minor`, `patch`, or a specific SemVer version number. And where `<push-remote>` is the git remote for the upstream repository `embedded-graphics/simulator`.
```

Edit: The suggestion did get messed up by the included code block. You will need to merge this manually.

**Review by rfuest — CHANGES_REQUESTED** [review]

**Review by rfuest — APPROVED** [review]
LGTM. The suggestion about the git remote will also need to be added to all other release docs.

**Comment by jamwaffles** [comment]
Thanks! Yeah, I'll add that into the other ones now. Prepare for review groundhog day...

**Pull request merged**
